FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/mbedtls/mbedtls_freertos_port.c

```diff
@@ -0,0 +1,286 @@
+
+    socket = ( Socket_t ) ctx;
+
+    return ( int ) FreeRTOS_recv( socket, buf, len, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Creates a mutex.
+ *
+ * @param[in, out] pMutex mbedtls mutex handle.
+ */
+void mbedtls_platform_mutex_init( mbedtls_threading_mutex_t * pMutex )
+{
+    configASSERT( pMutex != NULL );
+
+    /* Create a statically-allocated FreeRTOS mutex. This should never fail as
+     * storage is provided. */
+    pMutex->mutexHandle = xSemaphoreCreateMutexStatic( &( pMutex->mutexStorage ) );
+    configASSERT( pMutex->mutexHandle != NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Frees a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @note This function is an empty stub as nothing needs to be done to free
+ * a statically allocated FreeRTOS mutex.
+ */
+void mbedtls_platform_mutex_free( mbedtls_threading_mutex_t * pMutex )
+{
+    /* Nothing needs to be done to free a statically-allocated FreeRTOS mutex. */
+    ( void ) pMutex;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to lock a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @return 0 (success) is always returned as any other failure is asserted.
+ */
+int mbedtls_platform_mutex_lock( mbedtls_threading_mutex_t * pMutex )
+{
+    BaseType_t mutexStatus = 0;
+
+    configASSERT( pMutex != NULL );
+
+    /* mutexStatus is not used if asserts are disabled. */
+    ( void ) mutexStatus;
+
+    /* This function should never fail if the mutex is initialized. */
+    mutexStatus = xSemaphoreTake( pMutex->mutexHandle, portMAX_DELAY );
+    configASSERT( mutexStatus == pdTRUE );
+
+    return 0;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to unlock a mutex.
+ *
+ * @param[in] pMutex mbedtls mutex handle.
+ *
+ * @return 0 is always returned as any other failure is asserted.
+ */
+int mbedtls_platform_mutex_unlock( mbedtls_threading_mutex_t * pMutex )
+{
+    BaseType_t mutexStatus = 0;
+
+    configASSERT( pMutex != NULL );
+    /* mutexStatus is not used if asserts are disabled. */
+    ( void ) mutexStatus;
+
+    /* This function should never fail if the mutex is initialized. */
+    mutexStatus = xSemaphoreGive( pMutex->mutexHandle );
+    configASSERT( mutexStatus == pdTRUE );
+
+    return 0;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to generate a random number.
+ *
+ * @param[in] data Callback context.
+ * @param[out] output The address of the buffer that receives the random number.
+ * @param[in] len Maximum size of the random number to be generated.
+ * @param[out] olen The size, in bytes, of the #output buffer.
+ *
+ * @return 0 if no critical failures occurred,
+ * MBEDTLS_ERR_ENTROPY_SOURCE_FAILED otherwise.
+ */
+int mbedtls_platform_entropy_poll( void * data,
```

Comment:
Is there a reason for keeping the Windows implementation logic in this separate function instead of just keeping it in the `mbedtls_hardware_poll` function? 
If this function is being kept, it would be good to use a `win` suffix in the name to indicate that it is Windows implementation

Reply:
The simple reason is that we just reused the existing port implementation as is.

Reply:
Both the functions can be used by the mbedTLS functions based on how we configure the library. I would keep the same name for the function as the mbedTLS library uses the same name internally.